src/Components/ArticlePage.jsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+
+const ArticlePage = () => {
+  const { id } = useParams();
+  console.log(useParams());
+  const [article, setArticle] = useState(null);
+
+  useEffect(() => {
+    fetch(`https://my-nc-news-yyln.onrender.com/api/articles/${id}`)
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
+        console.log(data.article);
+        setArticle(data.article);
+      });
+  }, [id]); // this fetches the article if the id changes
+
+  if (!article) {
+    return <div>Loading...</div>;
+  }
```

Comment:
This does work, but I personally would prefer if a value, which in this case is a state value is consistent with its data type. I would not expect article to be a falsy data type and used like this.

As an alternative I would recommend considering another state to tell you if the data has been fetched or is currently being fetched